src/routes/tasks.rs

```diff
@@ -59,7 +60,34 @@ pub async fn create(
         .map(|task| (Status::Created, Json(task)))
 }
 
-pub async fn show(conn: &mut Conn, task: Task) -> Result<Json<Task>, Error> {
+async fn refresh_metrics_if_needed(
+    task: Task,
+    db: Db,
+    client: Client,
+    crypter: &Crypter,
+) -> Result<Task, Error> {
+    if OffsetDateTime::now_utc() - task.updated_at <= Duration::from_secs(5 * 60) {
```

Comment:
A five minute refresh interval is unlikely to cause problems, but it'd be nice to have this be configurable.

Reply:
Hmm. I think this might require more extensive work/thought.

Suppose that we're at the expiration window, and refreshing metrics takes 1 second. During that 1 second interval, multiple requests can come in and kick off metrics updates. (This is why we observed the old metrics clobbered the database when refreshing the page too much).

If we're in a situation where we're tempted to bump the refresh interval, it's probably because getting metrics is proving overwhelming. However, this won't actually help because the user will probably keep spamming the browser refresh button and kicking off more refreshes.

I think we may instead want an emergency flag to turn off metrics updates entirely, if anything. Or we need locking during updates, or run metrics updates asynchronously, or something. Thoughts?

Reply:
The divviup-api->Janus metrics query should be much much cheaper now, because it's just reading a single row from the `task_upload_counters` table as opposed to the old, very expensive query. Nonetheless, you raise some good points. Much as we have a flag to disable updating the `task_upload_counters` table on the Janus side, I think we should at least have a flag to disable the metrics in divviup-api. If we deploy this and observe that the load on Janus is a problem, we can turn it off with the flag, and then investigate the more robust solutions you suggest.

Reply:
There is in fact little bit more room for error in the read query since it has to aggregate across all `ord` shards. It _should_ always be a cheap index scan, but well postgres has been full of surprises lately. In any case a disable flag sounds good.

Reply:
I've added an environment variable `ENABLE_UPLOAD_METRICS`, which defaults to true but can be disabled with `ENABLE_UPLOAD_METRICS=false`.